modules/core/export/modules/csv/src/Plugin/Action/LogQuantityCsv.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\farm_export_csv\Plugin\Action;
+
+use Drupal\Core\Action\Plugin\Action\EntityActionBase;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\TempStore\PrivateTempStoreFactory;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Action that exports a CSV file of log quantities.
+ *
+ * @Action(
+ *   id = "farm_export_csv:log_quantity",
+ *   action_label = @Translation("Export Quantities CSV"),
+ *   type = "log",
+ *   confirm_form_route_name = "entity.quantity.csv_form",
+ * )
```

Comment:
This is a bit of a clever way to implement this Log Quantity CSV export... I found that we could reuse the `entity.quantity.csv_form` entirely, all we need to do is save the `log.quantities` into the tempstore instead of the `log` itself. This means we only need to implement this single Action plugin (not using a deriver) and don't need to duplicate almost all of the logic from `EntityCsvActionForm`.

This could create a little of bit of weird UX though... When the user selects/checks logs and clicks the "Export Quantities CSV" action, it will take them to the same "Export a CSV of [X] quantities?" confirmation form as you would get when using "Export CSV" action from a quantities view. It will not mention anything of the logs they selected. It's a little odd, but really it's just elegant and might not be too confusing to users anyways. If you do click 'Cancel' it takes you back to the correct destination Log page, not Log Quantities.

The major downside I foresee with this approach is that it just might not scale and exporting many logs with many quantities might crash, even if we are able to add a Batch API export to the general `EntityCsvActionForm`. This is simply because loading all the log quantities during the action plugin's `executeMultiple()` step can't be included in a Batch operation. Ideally, the Batch export for Log Quantities would handle a few or even a single _log_ (not quantity) in each batch operation. In this case I think the solution is providing a new Export Action Form that handles this Log + Quantity export more elegantly. I think this could be added/changed in the future without breaking change to the Action config entity added here.